WalletWasabi/Models/SmartTransaction.cs

```diff
@@ -75,6 +78,121 @@ public SmartTransaction(Transaction transaction, Height height, uint256 blockHas
 
 		#endregion Constructors
 
+		public string ToLine()
+		{
+			// GetHash is also serialized, so file can be interpreted with our eyes better.
+			var builder = new StringBuilder(GetHash().ToString());
+			builder.Append(':');
+			builder.Append(Transaction.ToHex());
+			builder.Append(':');
+			builder.Append(Height.Value.ToString());
+			builder.Append(':');
+			if (BlockHash != null)
+			{
+				builder.Append(BlockHash);
+			}
+			builder.Append(':');
+			builder.Append(BlockIndex);
+			builder.Append(':');
+			builder.Append(CorrectLabel(Label));
+			builder.Append(':');
+			builder.Append(FirstSeenIfMempoolTime?.ToUnixTimeSeconds().ToString() ?? "");
+			builder.Append(':');
+			builder.Append(IsReplacement);
+
+			return builder.ToString();
+		}
+
+		private string CorrectLabel(string label)
+		{
+			return Guard.Correct(label).Replace(":", "", StringComparison.Ordinal).Trim();
+		}
+
+		public static SmartTransaction FromLine(string line, Network expectedNetwork)
```

Comment:
No, because of cross process compatibility.  

- Now, we have the asyncmutex that makes sure only one guy writes the file.
- When we add a tx, we keep appending lines to the end, and if there's duplication in the file, we do not care, because at every initialization these duplications are corrected.
- When we remove tx, then we read the whole file and remove the txs we want to remove. We don't load, nor dump everything.